packages/core/addon/transforms/base-metadata-transform.js

```diff
@@ -28,13 +27,27 @@ export default DS.Transform.extend({
    * @returns {Object|Array} Ember Object | Array of Ember Objects
    */
   deserialize(serialized) {
-    let metadataService = get(this, 'metadataService');
+    let namespace = null;
 
     if (isArray(serialized)) {
-      return serialized.map(dimension => metadataService.getById(get(this, 'type'), dimension));
+      return serialized.map(dimension => {
+        namespace = null;
+        if (dimension.includes('.')) {
+          const splitName = serialized.split('.');
+          namespace = splitName[0];
+          dimension = splitName[1];
+        }
+        return this.metadataService.getById(this.type, dimension, namespace);
+      });
     }
 
-    return metadataService.getById(get(this, 'type'), serialized);
+    if (serialized.includes('.')) {
+      const splitName = serialized.split('.');
+      namespace = splitName[0];
+      serialized = splitName[1];
+    }
+
+    return this.metadataService.getById(this.type, serialized, namespace);
   },
```

Comment:
Can we do something like:

```js
_lookupMetadata(name) {
  if (name.includes('.')) {
    const [namespace, baseName] = name.split('.');
    return this.metadataService.getById(this.type,  baseName, namespace);
  } 
 return this.metadataService.getById(this.type, name, null);
},

deserialize(serialized) {
  if (isArray(serialized)) {
    return serialized.map(name => this._lookupMetadata(name))
  }
  return this._lookupMetadata(name);
}

```

Reply:
Code is no longer repeated as the array behavior has been removed.